### PR TITLE
added BackgroundSubtractorCNT

### PR DESCRIPTION
**Pull request by lhelontra**
New background subtraction solution, very fast compared with mog2 or Knn. Good solution for low spec hardware. For more details, see: https://www.theimpossiblecode.com/blog/fastest-background-subtraction-opencv/

### CODE REVIEW

**Comment by sovrasov** [comment]
:+1:

File: modules/bgsegm/include/opencv2/bgsegm.hpp [hunk 1]
```diff
@@ -183,10 +183,50 @@ class CV_EXPORTS_W BackgroundSubtractorGMG : public BackgroundSubtractor
 @param decisionThreshold Threshold value, above which it is marked foreground, else background.
  */
 CV_EXPORTS_W Ptr<BackgroundSubtractorGMG> createBackgroundSubtractorGMG(int initializationFrames=120,
-                                                                        double decisionThreshold=0.8);                                  
+                                                                        double decisionThreshold=0.8);
 
 //! @}
 
+/** @brief Background subtraction based on counting.
```
Comment (alalek) [inline modules/bgsegm/include/opencv2/bgsegm.hpp]: Please put code below before this line above:
```
//! @}
```
It is required for proper documentation generation.

File: modules/bgsegm/src/bgfg_subcnt.cpp [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+#include <functional>
+
+namespace cv
+{
+namespace bgsegm
+{
+
+/** @brief Implementation of background subtraction based on counting.
+ *  About as fast as MOG2 on a high end system (benchmarked on )
+ *  More than twice faster than MOG2 on cheap hardware (benchmarked on Raspberry Pi3).
```
Comment (alalek) [inline modules/bgsegm/src/bgfg_subcnt.cpp]: This part of comment is a "duplication" of comment from .hpp file. Probably it is better to fix it or eliminate duplication.

BTW, this comment is not a part of auto-generated documentation.

File: modules/bgsegm/src/bgfg_subcnt.cpp [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+class BackgroundSubtractorCNTImpl: public BackgroundSubtractorCNT
+{
+public:
+    /**
+     * @brief BackgroundSubtractorCNTImpl
+     * @param stability number of frames with same pixel color to consider stable
+     * @param useHistory determines if we're giving a pixel credit for being stable for a long time
+     * @param maxStability maximum allowed credit for a pixel in history
+     * @param isParallel determines if we're parallelizing the algorithm
+     */
```
Comment (alalek) [inline modules/bgsegm/src/bgfg_subcnt.cpp]: This documentation comment should be moved into .hpp file for `createBackgroundSubtractorCNT` function.

**Comment by lhelontra** [comment]
@alalek Thanks for your suggestions.

**Comment by sagi-z** [comment]
Hi guys,
I'm the author of this code.
It is great that you're inserting my implementation here.

Just one request - please restore this line to the class documentation comment (deleted in the merge):
Algorithm by Sagi Zeevi

(see here : https://github.com/sagi-z/BackgroundSubtractorCNT/blob/master/bgsubcnt.h)

Thanks!

**Comment by alalek** [comment]
@sagi-z Patch: #1290 ([docs preview](http://pullrequest.opencv.org/buildbot/export/pr_contrib/1290/docs/de/dca/classcv_1_1bgsegm_1_1BackgroundSubtractorCNT.html#details))

**Comment by sagi-z** [comment]
@alalek It looks okay now. Thanks for the quick response.

**Pull request merged**